### ISSUE TITLE
Accessibility findings after using screen reader

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#24339]
From times to times I'm testing with the orca screen reader from https://app.vagrantup.com/inclusivedesign

Instead of keeping local notes I'll use this ticket to dump whatever I've found so far.

- [ ] TODO: consolidate with findings from https://github.com/nextcloud/server/issues/21443

- [ ] top banner set aria label to make it clear it's the "main navigation" instead of just "banner" ?
- files app
   - [ ] hidden checkbox "show grid view" tabbing appears very early when tabbing into main content, no aria label
   - [ ] change tab order to have breadcrumb show up before grid view button ?
   - [ ] invisible "expand sharing section" checkbox can be tabbed to in left sidebar navigation, should be made visible and aria-label more explicit
   - [ ] "Home link" + breadcrumbs for home
      - might need better description instead of just "$name link". it is not clear that we are in the breadcrumb zone when entering.
      - or make it a role=list and role=listitem so at least it's clear that it's a list ? or is it a navigation ?
   - [ ] "New link" for new, need better description, and it's a push button, might need to override role to say "pushbutton"
   - [ ] sort links in table need better label instead of just "Name link", maybe make it a list ?
   - [ ] move focus to sidebar after clicking "Details" from the context menu
   - [ ] keep row hover style while tabbing to actions
- files_sharing
   - [ ] share tab entries appear as multiple lists, maybe should be only a single list (use role attributes for that)

Some issues specific to nextcloud vue components were raised here: https://github.com/nextcloud/nextcloud-vue/issues/1600

FYI @nextcloud/accessibility

**Comment by skjnldsv** [comment]
> * [ ]  hidden checkbox "show grid view" tabbing appears very early when tabbing into main content, no aria label

Aria labels are not needed if they are properly implemented.
Html semantics are always prefered to manual aria labelling.

Though the way the label is done is wrong as this should be as plain text inside as far as I know
https://github.com/nextcloud/server/blob/6f8147b18fadd749b6060ab540292c9ac6f85d61/apps/files/templates/index.php#L8-L9

**Comment by PVince81** [comment]
yeah, usually when I say "aria-label" I mean "whatever makes the screen reader say thing". could also be a hidden-visually or another approach

**Comment by susnux** [comment]
Resolved with our a11y refactoring with Nextcloud 30.
